File: code_challenge_solution/c#/200 Tic-tac-toe.cs
```csharp
using System;

namespace TicTacToe
{
    class Program
    {
        static char[] board = { '1', '2', '3', '4', '5', '6', '7', '8', '9' };
        static int player = 1; // By default Player 1 starts
        static int choice; // User choice for position

        static int flag = 0; // Flag for invalid input check

        // Check if the position entered by user is already filled or not
        // using the ASCII value of characters
        static void CheckPosition(char charToCheck, int position)
        {
            if (board[position] != 'X' && board[position] != 'O')
            {
                board[position] = charToCheck;
                player++;
                flag = 0;
            }
            else
            {
                Console.WriteLine("Sorry the row {0} is already marked with an {1}. Please choose another position!", position, board[position]);
                Console.WriteLine("\n");
                flag = 1;
            }
        }

        // Chech if any player has won the match or not
        static int CheckWin()
        {
            #region Horzontal Winning Condtion
            // Winning Condition For First Row
            if (board[0] == board[1] && board[1] == board[2])
            {
                return 1;
            }
            // Winning Condition For Second Row
            else if (board[3] == board[4] && board[4] == board[5])
            {
                return 1;
            }
            // Winning Condition For Third Row
            else if (board[6] == board[7] && board[7] == board[8])
            {
                return 1;
            }
            #endregion

            #region Vertical Winning Condtion
            // Winning Condition For First Column
            else if (board[0] == board[3] && board[3] == board[6])
            {
                return 1;
            }
            // Winning Condition For Second Column
            else if (board[1] == board[4] && board[4] == board[7])
            {
                return 1;
            }
            // Winning Condition For Third Column
            else if (board[2] == board[5] && board[5] == board[8])
            {
                return 1;
            }
            #endregion

            #region Diagonal Winning Condition
            else if (board[0] == board[4] && board[4] == board[8])
            {
                return 1;
            }
            else if (board[2] == board[4] && board[4] == board[6])
            {
                return 1;
            }
            #endregion

            #region Checking For Draw
            // Check if all field values have been set
            else if (board[0] != '1' && board[1] != '2' && board[2] != '3' &&
                     board[3] != '4' && board[4] != '5' && board[5] != '6' &&
                     board[6] != '7' && board[7] != '8' && board[8] != '9')
            {
                return -1;
            }
            #endregion

            else
            {
                return 0;
            }
        }

        // Print the current board status
        static void PrintBoard()
        {
            Console.WriteLine("|{0}|{1}|{2}|", board[0], board[1], board[2]);
            Console.WriteLine("|{0}|{1}|{2}|", board[3], board[4], board[5]);
            Console.WriteLine("|{0}|{1}|{2}|", board[6], board[7], board[8]);
        }

        static void Main(string[] args)
        {
            Console.WriteLine("Tic-Tac-Toe Game");
            do
            {
                Console.Clear(); // Whenever loop will be again start then screen will be clear
                Console.WriteLine("Player 1: X and Player 2: O");
                Console.WriteLine("\n");
                if (player % 2 == 0)
                {
                    Console.WriteLine("Turn Player 2");
                }
                else
                {
                    Console.WriteLine("Turn Player 1");
                }
                Console.WriteLine("\n");
                PrintBoard(); // calling the board Function
                Console.WriteLine("\n");
                // Check if any player has won the match or not
                int condition = CheckWin();

                if (condition == 1)
                {
                    Console.WriteLine("\n");
                    Console.WriteLine("Player {0} has won the game!", (player % 2) + 1);
                    break;
                }
                else if (condition == -1)
                {
                    Console.WriteLine("\n");
                    Console.WriteLine("The game is a draw!");
                    break;
                }

                // Checking valid position entered or not
                do
                {
                    flag = 0;
                    // Read the position and check if it is valid or not
                    if (player % 2 == 0)
                    {
                        Console.Write("Player 2, enter your position (1-9): ");
                    }
                    else
                    {
                        Console.Write("Player 1, enter your position (1-9): ");
                    }
                    choice = Convert.ToInt32(Console.ReadLine()) - 1;

                    if (choice >= 0 && choice < 9)
                    {
                        // Checking availability of the slot
                        CheckPosition((player % 2 == 0) ? 'O' : 'X', choice);
                    }
                    else
                    {
                        Console.WriteLine("Please enter a valid position (1-9)!");
                        Console.WriteLine("\n");
                        flag = 1;
                    }
                } while (flag != 0); // flag value will be changed if the position value entered is valid
            } while (flag != 1); // Loop will run until all possible positions are marked

            Console.WriteLine("\n");
            Console.WriteLine("Press any key to exit.");
            Console.ReadKey();
        }
    }
}
```

File: code_challenge_solution/python/200 Tic-tac-toe.py
```python
# Creating the Tic-tac-toe game

# Create a 3x3 board
board = [[' ' for _ in range(3)] for _ in range(3)]

# Function to print the board
def print_board():
    for row in board:
        print(' | '.join(row))
        print('---------')

# Function to check if there is a winning combination
def check_winner(symbol):
    # Check rows
    for row in board:
        if all(cell == symbol for cell in row):
            return True
    
    # Check columns
    for col in range(3):
        if all(board[i][col] == symbol for i in range(3)):
            return True
    
    # Check diagonals
    if board[0][0] == symbol and board[1][1] == symbol and board[2][2] == symbol:
        return True
    if board[0][2] == symbol and board[1][1] == symbol and board[2][0] == symbol:
        return True
    
    return False

# Function to check if the board is full
def is_board_full():
    for row in board:
        if ' ' in row:
            return False
    return True

# Function to play the game
def play_game():
    # Variable to keep track of the current turn
    turn = 'X'
    
    # Variable to keep count of the number of turns played
    num_turns = 0
    
    while True:
        # Print the board
        print_board()
        
        # Get the row and column from the user
        row = int(input('Enter the row (0-2): '))
        col = int(input('Enter the column (0-2): '))
        
        # Check if the selected position is empty
        if board[row][col] == ' ':
            # Update the board with the current symbol
            board[row][col] = turn
            
            # Increment the number of turns played
            num_turns += 1
            
            # Check if the current player has won
            if check_winner(turn):
                print_board()
                print(f'Player {turn} wins!')
                break
            
            # Check if the board is full and there is no winner
            if num_turns == 9:
                print_board()
                print('It\'s a tie!')
                break
            
            # Switch the turn to the other player
            turn = 'O' if turn == 'X' else 'X'
        else:
            print('Invalid move! Try again.')


# Start the game
play_game()
```
